applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -517,6 +517,26 @@ static void gnss_event_handler(int event)
 	}
 }
 
+static int do_cloud_send_msg(const char *message, int len)
+{
+	int err;
+
+	strcpy(rsp_buf, message);
```

Comment:
How long can `message` be, can it always fit into `rsp_buf`? From what I see it comes from "datamode" data, perhaps it'd be safe to use `strncpy` and verify length instead?

Reply:
It's simple test facility that we don't expect long JSON message. Size of rsp_buf is 2700 by default. The reason to use this buffer is nrf_cloud lib requires static storage for later access and a buffer in stack would cause issue after released.
Use `strcpy` to include NULL terminator, but `strncpy` should also work as we do tell nrf_cloud the length.

Reply:
i don't think the copy is even necessary, just assign `.buf = message,`
do you need to save the message in `rsp_buf` for other purposes?

if it is needed, a check for `len < rsp_buf` and/or using `strncpy` would be good here.

Reply:
It's protection that the buffer would not be released before sending out to cloud.
Added length check before copying.